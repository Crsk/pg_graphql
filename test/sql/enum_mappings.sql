begin;
create type my_enum as enum ('test', 'valid value');
comment on type my_enum is E'@graphql({"mappings": {"valid value": "valid_value"}})';
create table enums (
   id serial primary key,
   value my_enum
);
insert into enums (value) values ('test'), ('valid value');
select graphql.resolve($$
    {
      enumsCollection {
        edges {
            node {
             value
            }
        }
      }
    }
$$);
rollback;
