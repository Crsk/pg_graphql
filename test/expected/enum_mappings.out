begin;
create type my_enum as enum ('test', 'valid value', 'another value');
comment on type my_enum is E'@graphql({"mappings": {"valid value": "valid_value", "another value": "another_value"}})';
create table enums (
   id serial primary key,
   value my_enum
);
-- Seed with value that's valid in both Postgres and GraphQL
insert into enums (value) values ('test');
-- Mutation to insert
select graphql.resolve($$
mutation {
  insertIntoEnumsCollection(objects: [ { value: "valid_value" } ]) {
      affectedCount
  }
}
$$);
                            resolve                            
---------------------------------------------------------------
 {"data": {"insertIntoEnumsCollection": {"affectedCount": 1}}}
(1 row)

-- Mutation to update
select graphql.resolve($$
mutation {
  updateEnumsCollection(set: { value: "another_value" }, filter: { value: {eq: "test"} } ) {
    records { value }
  }
}
$$);
                                    resolve                                     
--------------------------------------------------------------------------------
 {"data": {"updateEnumsCollection": {"records": [{"value": "another_value"}]}}}
(1 row)

--- Query
select graphql.resolve($$
    {
      enumsCollection {
        edges {
            node {
             value
            }
        }
      }
    }
$$);
                                                       resolve                                                        
----------------------------------------------------------------------------------------------------------------------
 {"data": {"enumsCollection": {"edges": [{"node": {"value": "another_value"}}, {"node": {"value": "valid_value"}}]}}}
(1 row)

rollback;
